### PR TITLE
osu-lazer: 2022.1101.0 -> 2022.1228.0

### PR DESCRIPTION
**Pull request by delan**
###### Description of changes

Updated with `update.sh` in the package directory, but only tested on NixOS 22.05 because I haven’t upgraded yet. Release notes here but not exhaustive: [2022.1227.0](https://github.com/ppy/osu/releases/tag/2022.1227.0), [2022.1228.0](https://github.com/ppy/osu/releases/tag/2022.1228.0).

@oxalica @thiagokokada:

1. Could you please check if this works on 22.11 or 23.05, optionally removing the dotnet-sdk and dotnet-runtime lines that are no longer needed?
2. What do you think about switching to or providing a version that wraps the official AppImage (see below)? osu! lazer can now submit scores, but only if we use the AppImage, otherwise we get “invalid client hash” errors in network.log (ppy/osu#12491).

```nix
{ pkgs ? import <nixpkgs> {} }:
pkgs.appimageTools.wrapType2 {
  name = "osu";
  src = pkgs.fetchurl {
    url = "https://github.com/ppy/osu/releases/download/2022.1227.0/osu.AppImage";
    sha256 = "qawPm52P7dLFJlV3mNKuoEZTHjSdbkvGTg5I3kLumic=";
  };
  extraPkgs = pkgs: with pkgs; [ icu ];
}
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux (**22.05 only**)
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] ~~For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))~~
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by thiagokokada** [comment]
> What do you think about switching to or providing a version that wraps the official AppImage (see below)?

I would prefer to keep building from source, it is important for reproducibility reasons. But if we have an alternative `osu-lazer-bin` that builds from the AppImage, I don't see any issue.

File: pkgs/games/osu-lazer/default.nix [hunk 1]
```diff
@@ -10,22 +10,29 @@
...
+    sha256 = "sha256-0iKv7H1YbTsFkKopMp9iF52mfdGaK3idurtKPNFQmGs=";
   };
 
   projectFile = "osu.Desktop/osu.Desktop.csproj";
   nugetDeps = ./deps.nix;
 
+  # TODO remove in NixOS 22.11
+  dotnet-sdk = dotnetCorePackages.sdk_6_0;
+  dotnet-runtime = dotnetCorePackages.runtime_6_0;
+
```
Comment (thiagokokada) [inline pkgs/games/osu-lazer/default.nix:35]:
```suggestion

```

File: pkgs/games/osu-lazer/default.nix [hunk 2]
```diff
@@ -10,22 +10,29 @@
...
-    sha256 = "sha256-o0QHb8ebXKG0OZaYA9P/0OuOiSml3slYQ7gjyuP/ZyY=";
+    sha256 = "sha256-0iKv7H1YbTsFkKopMp9iF52mfdGaK3idurtKPNFQmGs=";
   };
 
   projectFile = "osu.Desktop/osu.Desktop.csproj";
   nugetDeps = ./deps.nix;
 
+  # TODO remove in NixOS 22.11
+  dotnet-sdk = dotnetCorePackages.sdk_6_0;
+  dotnet-runtime = dotnetCorePackages.runtime_6_0;
```
Comment (thiagokokada) [inline pkgs/games/osu-lazer/default.nix:34]:
```suggestion
```

File: pkgs/games/osu-lazer/default.nix [hunk 3]
```diff
@@ -10,22 +10,29 @@
...
 , dotnetCorePackages
+, libglvnd
+, xorg
+, udev
 }:
 
 buildDotnetModule rec {
   pname = "osu-lazer";
-  version = "2022.1101.0";
+  version = "2022.1227.0";
```
Comment (thiagokokada) [inline pkgs/games/osu-lazer/default.nix:20]: There is the `2022.1228.0` bugfix release, so I think we can change to it.

**Review by thiagokokada — CHANGES_REQUESTED** [review]
Just small nits and please squash your commits in one.

**Comment by thiagokokada** [comment]
> Could you please check if this works on 22.11 or 23.05, optionally removing the dotnet-sdk and dotnet-runtime lines that are no longer needed?

Checked and works fine. The `dotnet-sdk` and `dotnet-runtime` will not be needed.

File: pkgs/games/osu-lazer/default.nix (hunk 3 above)
Comment (delan) [reply pkgs/games/osu-lazer/default.nix]: done in 935bed0834734bc59ca1b04b3405edec51e853cb

File: pkgs/games/osu-lazer/default.nix (hunk 2 above)
Comment (delan) [reply pkgs/games/osu-lazer/default.nix:34]: done in 7b28118e8509545887c84129849663d6495a162d

**Comment by delan** [comment]
Thanks @thiagokokada!

> I would prefer to keep building from source, it is important for reproducibility reasons. But if we have an alternative `osu-lazer-bin` that builds from the AppImage, I don't see any issue.

I agree, leaving the AppImage for a separate `osu-lazer-bin` (at some point) sounds good.

> Just small nits and please squash your commits in one.

Repo settings don’t let me force push and I’m also not a committer, so I think I need your help for that.

**Comment by thiagokokada** [comment]
> I agree, leaving the AppImage for a separate osu-lazer-bin (at some point) sounds good.

Would you like to work on it @delan ?

**Comment by delan** [comment]
> Would you like to work on it @delan ?

Yeah, working on it now, I’ll open a separate init PR for it once I’m done.

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 208019` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>osu-lazer</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

**Comment by thiagokokada** [comment]
Thanks @delan.

**Pull request merged**